stepactions/reduce-snapshot/reduce-snapshot.yaml

```diff
@@ -0,0 +1,88 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: StepAction
+metadata:
+  name: reduce-snapshot-action
+spec:
+  image: quay.io/konflux-ci/appstudio-utils:ab6b0b8e40e440158e7288c73aff1cf83a2cc8a9@sha256:24179f0efd06c65d16868c2d7eb82573cce8e43533de6cea14fec3b7446e0b14
+  params:
+    - name: IMAGES
+      type: string
+    - name: SINGLE_COMPONENT
+      type: string
+    - name: CR_TO_QUERY
+      type: string
+    - name: CR_TO_QUERY_NAMESPACE
+      type: string
+      default: ""
+    - name: SNAPSHOT_PATH
+      type: string
+  env:
+    - name: SNAPSHOT
+      value: $(params.IMAGES)
+    - name: SINGLE_COMPONENT
+      value: $(params.SINGLE_COMPONENT)
+    - name: CR_TO_QUERY
+      value: $(params.CR_TO_QUERY)
+    - name: CR_TO_QUERY_NAMESPACE
+      value: $(params.CR_TO_QUERY_NAMESPACE)
+    - name: SNAPSHOT_PATH
+      value: $(params.SNAPSHOT_PATH)
+  results:
+    - name: snapshotCreationComponent
+      description: Name of C
+  script: |
+    #!/usr/bin/env bash
+    set -eux
+
+    echo "Single Component mode? ${SINGLE_COMPONENT}"
+    if [ "${SINGLE_COMPONENT}" == "true" ]; then
+    
+      CR_NAMESPACE_ARG=""
+      if [ ! -z "${CR_TO_QUERY_NAMESPACE}" ]; then
+        CR_NAMESPACE_ARG="-n ${CR_TO_QUERY_NAMESPACE}"
+      fi
+      SNAPSHOT_CREATION_TYPE=$(oc get "$CR_TO_QUERY" ${CR_NAMESPACE_ARG} -ojson \
+          | jq -rec '.metadata.labels."test.appstudio.openshift.io/type" // ""')
```

Comment:
Also, what is the `-e` for? You have a default value, so the jq command will never fail even if the field does not exist. Or do you anticipate the value being `null` or `false`? That's the only way it would fail.

Reply:
removed `-ec`